src/dune_rpc_impl/decl.mli

```diff
@@ -1,18 +0,0 @@
-open Dune_rpc_private
-open Dune_rpc_private.Decl
```

Comment:
Why is it necessary to move all this private stuff to the dune-rpc?

Reply:
With this model, the dune-rpc needs to be aware of these procedures to be able to be able to declare them (and their versions) as possible inputs.

Strictly speaking, I guess they don't *have* to be here; we could expose a separate mechanism for private procedures. I do think that it's possibly a good idea to have all the `Decl` objects in a centralized place regardless.

Reply:
> Strictly speaking, I guess they don't have to be here; we could expose a separate mechanism for private procedures

Perhaps I'm not fully understanding the code. Why do we need a separate mechanism? Isn't it enough to just extend the menu on the private side?

> I do think that it's possibly a good idea to have all the Decl objects in a centralized place regardless.

Yes, that would be fine but it imposes an annoying restriction on us. In dune-rpc, we cannot freely use any dependencies (I'm working on removing stdune for example). While if we defined the additional calls in the separate library, we'd be free to use whatever types/functions we want.

Reply:
>  I guess they don't have to be here; we could expose a separate mechanism for private procedures.

Perhaps I'm not understanding the code. Why is a separate mechanism needed? Isn't it enough to be able to define  a private menu and compose it with the public one?

> I do think that it's possibly a good idea to have all the Decl objects in a centralized place regardless.

That would be good if it didn't impose the following limitation on us. In dune-rpc, we have a very small dependency profile (i'm working on removing stdune from the dep list for example). We don't want to impose that limitation on our private methods. We should be able to use whatever types/functions we want.

Reply:
> Why is a separate mechanism needed? Isn't it enough to be able to define a private menu and compose it with the public one?

I guess "being able to define a private menu and compose it with the public one" is the mechanism I was thinking of. I'll work on adding this.

> That would be good if it didn't impose the following limitation on us. In dune-rpc, we have a very small dependency profile (i'm working on removing stdune from the dep list for example). We don't want to impose that limitation on our private methods. We should be able to use whatever types/functions we want.

This makes a lot of sense. However, this leads to another concern, in that the implementation of the versioning handler uses `Map`, `Table` and `Univ_map` from stdune; should I rewrite the code to avoid this?

Reply:
> This makes a lot of sense. However, this leads to another concern, in that the implementation of the versioning handler uses Map, Table and Univ_map from stdune; should I rewrite the code to avoid this?

You don't have to do it in this PR - it's already in this PR. But that will need to be done before 3.0